addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/discovery/NetatmoModuleDiscoveryService.java

```diff
@@ -127,31 +125,21 @@ private void discoverWelcomeHome() {
         }
     }
 
-    private boolean checkAlreadyExists(String id) {
-        return netatmoBridgeHandler.getThing().getThings().stream()
-                .filter(thing -> thing.getHandler() instanceof AbstractNetatmoThingHandler)
-                .map(device -> device.getProperties().get(EQUIPMENT_ID)).filter(Objects::nonNull)
-                .anyMatch(objId -> objId.equalsIgnoreCase(id));
-    }
-
     private void onDeviceAddedInternal(String id, String parentId, String type, String name, Integer firmwareVersion) {
-        // Prevent from adding already known devices
-        if (!checkAlreadyExists(id)) {
-            ThingUID thingUID = findThingUID(type, id);
-            Map<String, Object> properties = new HashMap<>();
-
-            properties.put(EQUIPMENT_ID, id);
-            if (parentId != null) {
-                properties.put(PARENT_ID, parentId);
-            }
-            if (firmwareVersion != null) {
-                properties.put(Thing.PROPERTY_VENDOR, "Netatmo");
-                properties.put(Thing.PROPERTY_FIRMWARE_VERSION, firmwareVersion);
-                properties.put(Thing.PROPERTY_MODEL_ID, type);
-                properties.put(Thing.PROPERTY_SERIAL_NUMBER, id);
-            }
-            addDiscoveredThing(thingUID, properties, name);
+        ThingUID thingUID = findThingUID(type, id);
+        Map<String, Object> properties = new HashMap<>();
+
+        properties.put(EQUIPMENT_ID, id);
+        if (parentId != null) {
+            properties.put(PARENT_ID, parentId);
+        }
+        if (firmwareVersion != null) {
+            properties.put(Thing.PROPERTY_VENDOR, "Netatmo");
+            properties.put(Thing.PROPERTY_FIRMWARE_VERSION, firmwareVersion);
+            properties.put(Thing.PROPERTY_MODEL_ID, type);
+            properties.put(Thing.PROPERTY_SERIAL_NUMBER, id);
         }
+        addDiscoveredThing(thingUID, properties, name);
     }
 
     private void addDiscoveredThing(ThingUID thingUID, Map<String, Object> properties, String displayLabel) {
```

Comment:
We should add `withRepresentationProperty(EQUIPMENT_ID)` to the `DiscoveryResultBuilder` statement. Shouldn't we?

Reply:
My tests shows that it works without.